### ISSUE TITLE
PieChart: Using MYSQLi to build my data values

### ISSUE THREAD

**Issue by hgpit** [kevinkhill/lavacharts#179]
What version are you having trouble with?
3.06

Hello,

I have just downloaded Lavacharts and am having a play with some of the examples.

I am trying to pass some simple values from a MySQL resultset to the pie chart builder but this seems to break the resulting pie chart.

For instance, not using MySQL data works:
```
$lava = new \Khill\Lavacharts\Lavacharts;
$OpenManifestStatuses = $lava->DataTable();
$OpenManifestStatuses->addStringColumn('Status')
                     ->addNumberColumn('Count');
$OpenManifestStatuses->addRow(['committed', 2])
                     ->addRow(['uncommitted', 1]);

$lava->PieChart('OpenManifestStatuses',$OpenManifestStatuses, [
    'title' => 'Open Manifest Statuses', 'is3D'   => true
]);
            <div id="manchart-div"></div>
            <?php echo $lava->render('PieChart', 'OpenManifestStatuses', 'manchart-div') ?>
```
So that looks good, but when I try to create it from my MySQL result set, it just generates a pie chart with 75% of the pie missing!

The difference being that I build the row data through a loop:
```
$vSQL = "SELECT 
        sr.status_description,
        COUNT(*) AS 'count'	
        FROM manifests m
        JOIN status_refs sr ON sr.status_ref = m.status_ref AND table_name = 'manifests'
        WHERE m.status_ref < 11
        GROUP BY m.status_ref";
$rQ = mysqli_query($vDB,$vSQL);

$OpenManifestStatuses = $lava->DataTable();
$OpenManifestStatuses->addStringColumn('Status')
                     ->addNumberColumn('Count');

while ( $aQR = mysqli_fetch_assoc($rQ) ) {
        $OpenManifestStatuses->addRow([$aQR['status_description'],$aQR['count']]);
}
```

Can someone please help me get this working properly? I am rubbish with OO PHP so it's probably down to not understanding how to push my row info within the loop.

Thank you!

**Comment by kevinkhill** [comment]
The values might be falling below the [default slice visibility threshold](https://developers.google.com/chart/interactive/docs/gallery/piechart#slice-visibility-threshold)?

Try adding that to your options and making the threshold a smaller amount? (I don't know the values you are working with)

**Comment by kevinkhill** [comment]
Or, maybe format your rows in the while loop, and build an array of row arrays. Then you can check the resulting array (debugger, var_dump, print_r) and see that is the proper format
```
[
    ['string', int],
    ['string', int],
    ['string', int],
    ...
]
```

Then pass that whole array to ```$OpenManifestStatuses->addRows()```

**Comment by hgpit** [comment]
Thank you very much for your reply, Kevin. I will have a look at your suggestions now.

The values which are currently returning from my data set are very low. The examples I gave in my post were actually real values (i.e. two pie segments, with values 1 and 2.

**Comment by hgpit** [comment]
Hi Kevin,

I think I've got to the bottom of it, and the solution may be useful to others using your excellent lavacharts.

I managed to get it working by converting my MySQL numerical results to decimal using the appropriate PHP function. So my working loop looks like this:

while ( $aQR = mysqli_fetch_assoc($rQ) ) {
    $OpenManifestStatuses->addRow([$aQR['status_description'],floatval($aQR['count'])]);
}

So, even though MySQL is returning an integer, I believe the MYSQLi PHP function returns all data values as strings - hence the need to convert them to float (or int also works) for lavacharts to play nice with them.
